fbossci-aws-lambas/us-east-1/github-status-webhook-handler/lambda_function.py

```diff
@@ -42,6 +46,26 @@ def handle_commits(commits, ref) -> None:
     print(f"Updated commit index for {branch_name}")
 
 
+def fetch_json(url):
+    headers = {'Accept': 'application/vnd.github.v3+json'}
+    token = os.environ.get('gh_pat')
+    if token:
+        headers['Authorization'] = f'token {token}'
+    with urlopen(Request(url, headers=headers)) as data:
+        return json.load(data)
+
+
+def get_workflow_name(job_id):
```

Comment:
how many URL requests do we anticipate with this approach per PR? seems like it is going to be same order of magnitude to amount of CI pipelines?
can we use https://docs.github.com/en/rest/reference/checks#list-check-runs-for-a-git-reference? (one API request should get all checks for a PR?)

Reply:
I'm not sure how to do that since this lambda gets triggered per `check_run` event:

https://github.com/pytorch/test-infra/blob/8b3adaf12626394134db83fcc6a719bb946cad5a/fbossci-aws-lambas/us-east-1/github-status-webhook-handler/lambda_function.py#L69-L76